Specific advice in unnecessary_nesting_linter metadata (#2759)
r-lib/lintr: NEWS.md

## Notes

* `expect_lint_free()` and other functions that rely on the {testthat} framework now have a consistent error message. (#2585, @F-Noelle).

> * `unnecessary_nesting_linter()` gives a more specific lint message identifying which "exit call" is unmatched & thus leading to the recommendation to reduce nesting (#2316, @MichaelChirico).

Reviewer: I found this sentence too convoluted (esp. the second principle phrase). How about?

```suggestion
* `unnecessary_nesting_linter()` gives a more specific lint message, identifying the unmatched "exit call" that prompts the recommendation to reduce nesting (#2316, @MichaelChirico).
```

Author: excellent, thanks!
